skyportal/tests/api/test_photometric_series.py

```diff
@@ -0,0 +1,2176 @@
+from sqlalchemy.exc import IntegrityError
+
+from skyportal.tests import api, assert_api, assert_api_fail
+from skyportal.models import DBSession, PhotometricSeries
+
+
+def convert_dataframe_to_bytes(df, metadata=None, encode=True):
+    """
+    Convert a pandas dataframe to bytes.
+
+    Parameters
+    ----------
+    df: pandas.DataFrame
+        The dataframe to convert to bytes.
+    metadata: dict
+        A dictionary of metadata to store in the HDF5 file.
+    encode: bool
+        Whether to encode the bytes as a base64 string.
+
+    Returns
+    -------
+    bytes:
+        The bytes of the dataframe.
+        If encode=True, the bytes are encoded as a base64 string
+        that can be sent over API calls.
+        If encode=False, the bytes are the raw bytes of the HDF5 file.
+    """
+
+    # this store should work without writing to disk
+    # if you open a regular store you'd just need
+    # to delete the file at the end
+    # ref: https://github.com/pandas-dev/pandas/issues/9246#issuecomment-74041497
+    with pd.HDFStore(
+        'test_file.h5', mode='w', driver="H5FD_CORE", driver_core_backing_store=0
+    ) as store:
+        store.put(
+            'phot_series',
+            df,
+            format='table',
+            index=None,
+            track_times=False,
+        )
+        if metadata is not None:
+            store.get_storer('phot_series').attrs.metadata = metadata
+
+        data = store._handle.get_file_image()
+
+        if encode:
+            data = base64.b64encode(data)
+
+    # should not be any file like this
+    assert not os.path.isfile('test_file.h5')
+
+    return data
+
+
+def test_hdf5_file_vs_memory_hash():
+    df = pd.DataFrame(
+        data=[[1, 2, 3, 4], [5, 6, 7, 8], [9, 10, 11, 12]], columns=['a', 'b', 'c', 'd']
+    )
+    mem_buf = convert_dataframe_to_bytes(df, encode=False)
+    mem_hash = hashlib.md5()
+    mem_hash.update(mem_buf)
+
+    # did not save the data to disk!
+    assert not os.path.isfile('test_string')
+
+    # make sure hashes saved at different times are the same
+    time.sleep(1)
+
+    # store the data on disk and check the hash of that
+    filename = 'try_saving_hdf5_file_with_hash.h5'
+    try:  # cleanup at end
+        with pd.HDFStore(filename, mode='w') as store:
+            store.put(
+                'phot_series',
+                df,
+                format='table',
+                index=None,
+                track_times=False,
+            )
+
+        with open(filename, 'rb') as f:
+            file_buf = f.read()
+            file_hash = hashlib.md5()
+            file_hash.update(file_buf)
+
+        # assert len(file_buf) == len(mem_buf)
+        assert file_hash.hexdigest() == mem_hash.hexdigest()
+
+        # make sure the same hash is made even when
+        # changing the versions of pandas, etc.
+        # if not, we will need to re-make all the hashes!
+        assert file_hash.hexdigest() == 'daf70e10284a36020af2cc102ae3d32a'
```

Comment:
this hash string might change with different hdf5 versions. might want to generate this dynamically.

Reply:
Yes, the idea is that when the version changes we would be made aware by this test failure. 
Such a change in version would require that we re-save and re-calculate the hash for all photometric series.